packages/aws-rfdk/lib/deadline/lib/version-query.ts

```diff
@@ -0,0 +1,156 @@
+
+  /**
+   * @inheritdoc
+   */
+  readonly abstract linuxFullVersionString: string;
+}
+
+/**
+ * This class encapsulates information about a particular version of Thinkbox's Deadline software.
+ * Information such as the version number, and where to get installers for that version from Amazon S3.
+ *
+ * The version of an official release of Deadline is always four numeric version components separated by dots.
+ * ex: 10.1.8.5. We refer to the components in this version, in order from left-to-right, as the
+ * major, minor, release, and patch versions. For example, Deadline version 10.1.8.5 is majorVersion 10, minorVersion 1,
+ * releaseVersion 8, and patchVersion 5.
+ *
+ * All of the installers provided by an instance of this class must be for the same Deadline release (ex: 10.1.8),
+ * but the patch versions may differ between operating systems depending on the particulars of that release of Deadline.
+ * This class provides a simple way to query a version of Deadline prior to or during deployment of a
+ * CDK app.
+ *
+ * You pass an instance of this class to various Deadline constructs in this library to tell those
+ * constructs which version of Deadline you want them to use, and be configured for.
+ */
+export class VersionQuery extends VersionQueryBase {
+  /**
+   * @inheritdoc
+   */
+  readonly majorVersion: number;
+
+  /**
+   * @inheritdoc
+   */
+  readonly minorVersion: number;
+
+  /**
+   * @inheritdoc
+   */
+  readonly releaseVersion: number;
+
+  /**
+   * @inheritdoc
+   */
+  readonly linuxFullVersionString: string;
+
+  /**
+   * @inheritdoc
+   */
+  readonly linuxInstallers: PlatformInstallers;
+
+  constructor(scope: Construct, id: string, props?: VersionQueryProps) {
+    super(scope, id);
+
+    const lambdaCode = Code.fromAsset(join(__dirname, '../..', 'core', 'lambdas', 'nodejs'));
+
+    const lambdaFunc = new SingletonFunction(this, 'VersionProviderFunction', {
+      uuid: '2e19e243-16ee-4d1a-a3c9-18d35eddd446',
+      description: 'Used by the Version construct to get installer locations for a specific Deadline version.',
+      code: lambdaCode,
+      runtime: Runtime.NODEJS_12_X,
+      handler: 'version-provider.handler',
+      timeout: Duration.seconds(30),
+      logRetention: RetentionDays.ONE_WEEK,
+    });
+
+    const deadlineProperties: IVersionProviderResourceProperties = {
+      versionString: props?.version,
+    };
+
+    const deadlineResource = new CustomResource(this, 'DeadlineResource', {
+      serviceToken: lambdaFunc.functionArn,
+      properties: deadlineProperties,
+      resourceType: 'Custom::RFDK_DEADLINE_INSTALLERS',
+    });
+
+    this.majorVersion = Token.asNumber(deadlineResource.getAtt('MajorVersion'));
+    this.minorVersion = Token.asNumber(deadlineResource.getAtt('MinorVersion'));
+    this.releaseVersion = Token.asNumber(deadlineResource.getAtt('ReleaseVersion'));
+
+    this.linuxFullVersionString = Token.asString(deadlineResource.getAtt('LinuxFullVersionString'));
```

Comment:
I tried this and got output like `-1.888154589708867e+289.-1.8881545897088673e+289.-1.8881545897088675e+289.-1.8881545897088678e+289`

Reply:
I recall having this issue embedding [number-encoded CDK tokens](https://docs.aws.amazon.com/cdk/latest/guide/tokens.html#tokens_number) in a string and this won't work naively. The CDK doesn't replace numeric tokens within a string. I suppose this may be because there is loss of precision when encoding floating point numbers as strings.

There is a work-around that involves first checking the number to see if it is unresolved (in this case we know it is an unresolved token since it comes from a resource output attribute), and if so, wrap that numeric token with a string token. You can see an example of this in our [`Endpoint.portAsString()`](https://github.com/aws/aws-rfdk/blob/79e0d1f86389b289596974b15bcf49c735d72e56/packages/aws-rfdk/lib/core/lib/endpoint.ts#L118-L133) implementation.

Reply:
Updated to use Josh's suggestion.